core/trino-main/src/test/java/io/trino/cost/TestFilterStatsCalculator.java

```diff
@@ -301,6 +311,78 @@ public void testAndStats()
 
         assertExpression("CAST(NULL AS boolean) AND CAST(NULL AS boolean)").equalTo(zeroStatistics);
         assertExpression("CAST(NULL AS boolean) AND (x < 0e0 AND x > 1e0)").equalTo(zeroStatistics);
+
+        Consumer<SymbolStatsAssertion> symbolAssertionX = symbolAssert -> symbolAssert.averageRowSize(4.0)
+                .lowValue(-5.0)
+                .highValue(5.0)
+                .distinctValuesCount(20.0)
+                .nullsFraction(0.0);
+        Consumer<SymbolStatsAssertion> symbolAssertionY = symbolAssert -> symbolAssert.averageRowSize(4.0)
+                .lowValue(1.0)
+                .highValue(5.0)
+                .distinctValuesCount(16.0)
+                .nullsFraction(0.0);
+
+        assertExpression(
+                "(x BETWEEN -5 AND 5) AND y > 1",
+                Session.builder(session).setSystemProperty(FILTER_CONJUNCTION_INDEPENDENCE_FACTOR, "0").build())
+                .outputRowsCount(375)
+                .symbolStats("x", symbolAssertionX)
+                .symbolStats("y", symbolAssertionY);
+
+        assertExpression(
```

Comment:
I've added these tests but the result turns out to be the same for independence factor 1 and 0.5 because stats normaliser has the below check
```
if (distinctValuesCount == 0.0) {
      return SymbolStatsEstimate.zero();
}
```

Reply:
I wonder if we can trigger a case where intersected stats have different `null` fraction